presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraRecordSink.java

```diff
@@ -41,37 +49,36 @@
 {
     private static final DateTimeFormatter DATE_FORMATTER = ISODateTimeFormat.date().withZoneUTC();
 
-    private final int fieldCount;
     private final CassandraSession cassandraSession;
-    private final String insertQuery;
+    private final PreparedStatement insert;
     private final List<Object> values;
     private final List<Type> columnTypes;
+    private final boolean generateUUID;
+
     private int field = -1;
 
-    @Inject
-    public CassandraRecordSink(CassandraOutputTableHandle handle, CassandraSession cassandraSession)
+    public CassandraRecordSink(
+            CassandraSession cassandraSession,
+            String schemaName,
+            String tableName,
+            List<String> columnNames,
+            List<Type> columnTypes,
+            boolean generateUUID)
     {
-        this.fieldCount = requireNonNull(handle, "handle is null").getColumnNames().size();
-        this.cassandraSession = requireNonNull(cassandraSession, "cassandraSession is null");
+        this.cassandraSession = requireNonNull(cassandraSession, "cassandraSession");
+        this.columnTypes = ImmutableList.copyOf(requireNonNull(columnTypes, "columnTypes is null"));
+        this.generateUUID = generateUUID;
 
-        String schemaName = handle.getSchemaName();
-        StringBuilder queryBuilder = new StringBuilder(String.format("INSERT INTO \"%s\".\"%s\"(", schemaName, handle.getTableName()));
-        queryBuilder.append("id");
-
-        for (String columnName : handle.getColumnNames()) {
-            queryBuilder.append(",").append(columnName);
+        Insert insert = insertInto(schemaName, tableName);
+        if (generateUUID) {
+            insert.value("id", bindMarker());
         }
-        queryBuilder.append(") VALUES (?");
-
-        for (int i = 0; i < handle.getColumnNames().size(); i++) {
-            queryBuilder.append(",?");
+        for (String columnName : columnNames) {
+            insert.value(columnName, bindMarker());
         }
-        queryBuilder.append(")");
+        this.insert = cassandraSession.executeWithSession(s -> s.prepare(insert));
```

Comment:
When this is rebased on https://github.com/prestodb/presto/pull/7914, how will this work? You'll need to add another method to CassandraSession, right?